### PR TITLE
[flutter_tools] stop using throttled print for service extensions

### LINKED ISSUE #33184: It is too easy to accidentally press 't' when performing a hot reload
Using Flutter from cli and trying to hot reload is hard because the 't' key is so close to the 'r' and accidently pressing on it results cli to try to print the widget tree and since this process is a long one every press end with frustratingly waiting like 2 mins every single time. And I couldn't find any info about this on how to disable it. Please add this as a feature or show me a way to avoid this.

### LINKED ISSUE #49143: Add a confirmation dialog for dumping the rendering tree to console
## Use case
Every other time I do a hot restart for Flutter web it happens that I hit the **T** key by mistake, which is next to the **R** key on the keyboards with US layout.
For complex projects, it take a while to dump everything and keeps me waiting.
I can choose to restart the project, but it takes time to restart it.

This happens just to the **Dump rendering tree to the console.** triggered by **T**, but it can happen with other console-invader commands.

## Proposal
One solution would be to add a confirmation dialog **Y/n** and every key except **Y** to be considered no.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Currently the service extensions that print a debug view of the widget, render object, semantics, or layer tree can lock a device for several seconds. This is due to the printing being driven entirely by the device side, and the printing being throttled to prevent logs from being dropped by the host system. 

This has inspired a good deal of frustration, as users that accidentally press these keys find their devices locked up and terminal filled with irrelevant information.

See:

Fixes https://github.com/flutter/flutter/issues/33184
Fixes https://github.com/flutter/flutter/issues/49143

https://stackoverflow.com/questions/56726247/how-to-disable-widget-tree-printing-in-flutter-cli-pressing-t-key-accidently

Remove the lockup by push all of the data over the vm service to the tool for printing. Makes `t`, `S`, `w` immediately print the entire buffer.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/binding.dart [hunk 1]
```diff
@@ -1027,20 +1030,18 @@ void runApp(Widget app) {
...
+/// Return a string representation of the currently running app.
+String debugDumpApp() {
   assert(WidgetsBinding.instance != null);
-  String mode = 'RELEASE MODE';
-  assert(() {
-    mode = 'CHECKED MODE';
-    return true;
-  }());
-  debugPrint('${WidgetsBinding.instance.runtimeType} - $mode');
+  final String mode = kDebugMode ? 'DEBUG MODE' : 'PROFILE MODE';
```
Comment (jonahwilliams) [inline packages/flutter/lib/src/widgets/binding.dart:1036]: We changed this to DEBUG MODE sometime in 2018, so just a minor cleanup here...

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/vmservice.dart [hunk 2]
```diff
@@ -515,49 +515,55 @@ extension FlutterVmService on vm_service.VmService {
...
+    return response['data']?.toString();
   }
 
-  Future<Map<String, dynamic>> flutterDebugDumpSemanticsTreeInInverseHitTestOrder({
+  Future<String> flutterDebugDumpSemanticsTreeInInverseHitTestOrder({
     @required String isolateId,
-  }) {
-    return invokeFlutterExtensionRpcRaw(
+  }) async {
+     final Map<String, Object> response = await invokeFlutterExtensionRpcRaw(
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/vmservice.dart:562]: nit: extra space in indent.

File: packages/flutter/lib/src/widgets/binding.dart [hunk 3]
```diff
@@ -387,11 +387,14 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
     super.initServiceExtensions();
 
     if (!kReleaseMode) {
-      registerSignalServiceExtension(
+
```
Comment (zanderso) [inline packages/flutter/lib/src/widgets/binding.dart:390]: nit: extra newline

**Review by zanderso — APPROVED** [review]
lgtm w/ nits

File: packages/flutter_tools/lib/src/vmservice.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/vmservice.dart]: Fixed

File: packages/flutter/lib/src/widgets/binding.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/widgets/binding.dart]: Fixed

**Pull request merged**